domain/segments.go

```diff
@@ -108,3 +109,15 @@ func toSDKSegments(segments *[]clientgen.Segment) []rest.Segment {
 
 	return results
 }
+
+func ConvertServingRulesToRules(servingRules []clientgen.GroupServingRule) []clientgen.Clause {
+	rules := make([]clientgen.Clause, 0, len(servingRules))
+
+	for _, s := range servingRules {
+		if len(s.Clauses) == 0 {
+			continue
+		}
+		rules = append(rules, s.Clauses[0])
```

Comment:
Is there only ever one clause?

Reply:
No, but this is the thing where a serving rule (AND rule) can have multiple AND clauses. But for older sdks that only understand the OR rule format we only send them the first clause from any AND rule. That's why we just pull out the first clause here then ignore the rest